io-engine/src/bdev/util/uri.rs

```diff
@@ -46,3 +46,9 @@ pub(crate) fn uuid(
 ) -> Result<Option<uuid::Uuid>, uuid::Error> {
     value.map(|uuid| uuid::Uuid::parse_str(&uuid)).transpose()
 }
+
+pub(crate) fn extract_nexus_name(nqn: &str) -> Option<String> {
+    let vec: Vec<String> = nqn.split(':').map(|s| s.to_string()).collect();
+
+    vec.get(1).cloned()
```

Comment:
Nit: don't need to clone two strings, just the "right" one:

```suggestion
    let vec: Vec<&str> = nqn.split(':').collect();

    vec.get(1).map(ToString::to_string)
```